eis_toolkit/conversions/csv_to_geopandas.py

```diff
@@ -0,0 +1,98 @@
+from typing import List
+
+import csv as reader
+import pandas as pd
+import geopandas
+
+from eis_toolkit.exceptions import (InvalidParameterValueException,
+                                    InvalidWktFormatException,
+                                    InvalidColumnIndexException)
+
+
+def _csv_to_geopandas(  # type: ignore[no-any-unimported]
+    csv: str,
+    indexes: List[int],
+    target_EPSG: int,
+) -> geopandas.GeoDataFrame:
+
+    with open(csv, 'r') as f:
+        has_header = reader.Sniffer().has_header(f.read(1024))
+
+    if has_header:
+        df = pd.read_csv(csv)
+        if len(indexes) == 1:
+            if len(df.columns) < indexes[0]:
+                raise InvalidColumnIndexException
+            column_names = []
+            for row in df:
+                column_names.append(row)
+            geom_column = column_names[indexes[0]]
+            try:
+                geoms = geopandas.GeoSeries.from_wkt(df[geom_column])
+                geodataframe = geopandas.GeoDataFrame(df, crs=target_EPSG, geometry=geoms)
+                return geodataframe
+            except:
+                raise InvalidWktFormatException
+
+        else:
+            if len(df.columns) < indexes[0] or len(df.columns) < indexes[1]:
+                raise InvalidColumnIndexException
+            column_names = []
+            for row in df:
+                column_names.append(row)
+            try:
+                geom_x = column_names[indexes[0]]
+                geom_y = column_names[indexes[1]]
+                geodataframe = geopandas.GeoDataFrame(
+                    df, crs=target_EPSG, geometry=geopandas.points_from_xy(df[geom_x], df[geom_y]))
+                return geodataframe
+            except:
+                raise InvalidParameterValueException
+    else:
+        df = pd.read_csv(csv, header=None)
+        if len(indexes) == 1:
+            if len(df.columns) < indexes[0]:
+                raise InvalidColumnIndexException
+            try:
+                geoms = geopandas.GeoSeries.from_wkt(df[indexes[0]])
+                geodataframe = geopandas.GeoDataFrame(df, crs=target_EPSG, geometry=geoms)
+                return geodataframe
+            except:
+                raise InvalidWktFormatException
+        else:
+            if len(df.columns) < indexes[0] or len(df.columns) < indexes[1]:
+                raise InvalidColumnIndexException
+            try:
+                geodataframe = geopandas.GeoDataFrame(
+                    df, crs=target_EPSG, geometry=geopandas.points_from_xy(df[indexes[0]], df[indexes[1]]))
+                return geodataframe
+            except:
+                raise InvalidParameterValueException
+
+
+def csv_to_geopandas(  # type: ignore[no-any-unimported]
+    csv: str,
```

Comment:
Is there a reason not to use `pathlib.Path` (`from pathlib import Path`) for paths here rather than passing the path as as a `str`? I believe it is already used elsewhere in `eis_toolkit`. It helps to avoid errors later with path separator differences between Windows and linux-based operating systems. Also less ambiguous as a `str` type can represent any strings rather than only file paths.

```suggestion
    csv: Path,
```

Reply:
Thank you for the comments. The reason I opted for using indexes rather than column names is so that the function could handle .csv files without headers. As for the general suggestions, I am implementing all of the suggested changes.